lua/lint.lua

```diff
@@ -103,16 +103,39 @@ function M.lint(linter, client_id)
     cwd = vim.fn.getcwd(),
     detached = true
   }
-  assert(linter.cmd, 'Linter definition must have a `cmd` set: ' .. vim.inspect(linter))
-  handle, pid_or_err = uv.spawn(linter.cmd, opts, function(code)
+  local cmd = linter.cmd
+  if type(cmd) == 'function' then
+    cmd = cmd(bufnr)
```

Comment:
On second thought, I have doubts about the use of passing the buffer number. I can't think of a case where you'd need to act on anything but the current buffer, if you even care about the explicit buffer number to begin with.

Reply:
Yes I also think it's not necessary. It can't be asynchronous, so the cmd function can always call `api.nvim_get_current_buf()` itself if it needs the current buffer number.